File: leetcode/217.ContainsDuplicate.go
package leetcode

func containsDuplicate(nums []int) bool {
	mappings := make(map[int]struct{})

	for _, num := range nums {
		if _, ok := mappings[num]; ok {
			return true
		}
		mappings[num] = struct{}{}
	}

	return false
}

File: leetcode/217.ContainsDuplicate.ts
function containsDuplicate(nums: number[]): boolean {
  const maps: Record<number, boolean> = {}
  for (const num of nums) {
    if (maps[num]) return true

    maps[num] = true
  }

  return false
}
console.log(containsDuplicate([1, 2, 3, 1]))
console.log(containsDuplicate([1, 2, 3, 4]))
console.log(containsDuplicate([1, 1, 1, 3, 3, 4, 3, 2, 4, 2]))
